sklearn/utils/_repr_html/tests/test_estimator.py

```diff
@@ -615,3 +615,19 @@ def test_estimator_html_repr_table():
     """Check that we add the table of parameters in the HTML representation."""
     est = LogisticRegression(C=10.0, fit_intercept=False)
     assert "parameters-table" in estimator_html_repr(est)
+
+
+def test_features_number():
+    X, y = load_iris(return_X_y=True)
+    categorical_transf = Pipeline(
+        steps=[
+            ("encoder", OneHotEncoder(handle_unknown="ignore")),
+        ]
+    )
+    categorical_transf.fit(X, y)
+    assert "123 Output features" in estimator_html_repr(categorical_transf)
+
+
+def test_features_number_not_included():
+    est = LogisticRegression(C=10.0, fit_intercept=False)
+    assert "Output features" not in estimator_html_repr(est)
```

Comment:
Let's create a `test_features.py` file to add those new test. I think that you need to update those tests. I think that we can test:

- Make sure that we don't show anything when the estimator is not fitted
- Make sure that we have have the boxes when the estimator is fitted
- Make sure that we have the right feature names in the table
- Make sure that it have the expected number of "X features" when having a column transformer in a pipeline
- Make sure that thing works even with `Transformer` that do not provide the full scikit-learn API. In short, we should add a `sklearn.utils._testing.MinimalTransformer` in a `Pipeline` and check that it does not crash.